automated-testing/shadow-testing/readme.md

```diff
@@ -0,0 +1,38 @@
+
+![Shadow Testing Overview](images/shadow-testing.png)
+
+ Referencing back to one of the [Principles of Chaos Engineering](https://principlesofchaos.org/), mentions importance of sampling real traffic like below:
+
+> Systems behave differently depending on environment and traffic patterns. Since the behavior of utilization can change at any time, sampling real traffic is the only way to reliably capture the request path. To guarantee both authenticity of the way in which the system is exercised and relevance to the current deployed system, Chaos strongly prefers to experiment directly on production traffic.
+
+With this Shadow Testing approach we're leveraging real customer behavior in V-Next environment with sampling real traffic and mitigating the risks which users may face on production.
+
+We're testing actual behavior of the product and this cause zero impact to production to test new features since traffic is replicated to V-next environment.
+
+There're some similarities with [Dark Launching](https://martinfowler.com/bliki/DarkLaunching.html), Dark Launching propose to integrate new feature into production release code and users are not able to see this feature, on the backend you can test your feature and improve the performance until it's acceptable. Also [Feature Toggle](https://martinfowler.com/bliki/FeatureToggle.html) approach is providing you an ability to enable/disable your new feature on the production on UI level, with this approach your new feature will be visible to users and you can collect feedback. Using Dark Launching with Feature Toggle approach can be very useful for introducing a new feature.
+
+## Shadow Testing Frameworks and Tools
+
+There are some tools to implement shadow testing, the most popular one is [Diffy](https://github.com/opendiffy/diffy). It was created and used at Twitter then original author and a former Twitter employee maintains their own version of this project, called [Opendiffy](https://github.com/opendiffy/diffy). Twitter announced this tool on the blog as "[Testing services without writing tests](https://blog.twitter.com/engineering/en_us/a/2015/diffy-testing-services-without-writing-tests.html)".
+
+ As of today Diffy is used in production by Twitter, Airbnb, Airbnb, Baidu and Bytedance companies.
+
+> Diffy finds potential bugs in your service using running instances of your new code and your old code side by side. Diffy behaves as a proxy and multicasts whatever requests it receives to each of the running instances. It then compares the responses, and reports any regressions that may surface from those comparisons. The premise for Diffy is that if two implementations of the service return “similar” responses for a sufficiently large and diverse set of requests, then the two implementations can be treated as equivalent and the newer implementation is regression-free.
```

Comment:
So Diffy does expect a response from V-Next? If so, this conflicts with what was mentioned earlier, could u plz fix that

Reply:
Thank you so much @alkofahi, yes we're using responses to compare with V-Current responses. I've updated previous mention and diagram as well. Thanks for highlighting this point.